### PR TITLE
steps definition for sidecar-pattern-jenkins feature

### PR DESCRIPTION
**Pull request by jitendar-singh**
- updating the scenario in sidecar-pattern-jenkins.feature file
- defining the steps for the feature in steps.py
- updating openshift.py
- updating the framework to use tables and improve the resuability
- updating the nojejs-sample pipeline, Version of NodeJS image to be used (14-ubi8, or latest).

### CODE REVIEW

File: smoke/features/steps/openshift.py [hunk 1]
```diff
@@ -214,6 +222,19 @@ def get_resource_info_by_jsonpath(self, resource_type, name, namespace, json_pat
...
+    def get_resources_info_by_jsonpath(self, resource_type, namespace, json_path, wait=False):
+        output, exit_code = self.cmd.run(f'oc get {resource_type} -n {namespace} -o "jsonpath={json_path}"')
+        if exit_code != 0:
+            if wait:
+                attempts = 5
+                while exit_code != 0 and attempts > 0:
+                    output, exit_code = self.cmd.run(
+                        f'oc get {resource_type} -n {namespace} -o "jsonpath={json_path}"')
+                    attempts -= 1
+                    time.sleep(5)
```
Comment (akram) [inline smoke/features/steps/openshift.py:235]: this one also should be parametrizable

File: smoke/features/steps/steps.py [hunk 2]
```diff
@@ -153,4 +151,65 @@ def jenkinsMasterPodStatus(context):
...
+    item = oc.search_resource_in_namespace('buildconfig', 'pipes', current_project)
+    if item is None:
+        raise AssertionError
+    else:
+        print(f'buildconfig {item} found')
+    
+@then(u'we check for "java-builder" agent node and "nodejs-builder" agent node are created')
+def check_pods(context):
+    current_project = context.current_project
+    time.sleep(75)
```
Comment (akram) [inline smoke/features/steps/steps.py:193]: that's a long sleep time, we need something more flexible and able to fail faster in case of an error.

**Review by akram — CHANGES_REQUESTED** [review]

**Comment by gabemontero** [comment]
/label docs-approved 

QE only

**Comment by gabemontero** [comment]
/label px-approved

QE only

**Comment by gabemontero** [comment]
/label qe-approved

QE centric

**Pull request merged**